proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,456 @@
+request. After review, all pull requests will follow the process and timelines
+defined in the specification.
+
+## Rationale/FAQ
+
+### Q. Why do we limit the number of in flight reviews?
+
+We do not want to discourage users from contributing or proposing changes, but
+we need to acknowledge that the maintainers have limited bandwidth to give
+feedback on implementation or design and review pull requests. The maintainers
+have work that comes from outside of the GitHub repository that needs to be
+prioritized along with user contributions. Controlling the volume of pull
+request reviews and prioritizing completing the in flight reviews over new
+reviews will result in a shorter time to merge and release a contributor's
+change for a feature requested by the community. It will also reduce the amount
+of context switching for the maintainers, allowing them to focus on giving high
+quality feedback to users. As we use this framework to process existing feature
+requests, we can revisit what the in flight limit should be.
+
+### Q. Why do we require issues to be opened instead of just pull requests?
+
+Having a defined process for intake reduces the burden on deciding which way is
+the best way. We're taking an opinionated stance that code changes made in
+isolation generally do not provide enough context on what problem exists and why
+the proposed change is the best way to fix it. Reviewing a pull request also
+requires significant effort on the part of maintainers - reading someone elses
+code can be an intensive task. Before dedicating that effort, we feel that
+having a more general discussion about the problem and solutions will help
+reduce the overall effort into code review.
+
+Pull requests are often made for problems that only affect the contributor or a
+very small portion of the user base. Requiring an issue provides a mechanism to
+request structured feedback in the form of "upvotes" or GitHub reactions to
+estimate the impact of the issue on the community.
+
+Logistically, using GitHub issues also centralizes the metadata on what work is
+available and ongoing via labels. While pull requests are really GitHub issues
+with code changes attached to them, it is difficult to interrogate issues and
+pull requests together to gather metrics, check on statuses and assignments, and
+perform automated tasks. Having all of the labels and data on a single entity
+type reduces the burden on maintaining the system.
+ 
+We do not intend this to be a gatekeeping or pointless task. If a user does open
+a pull request without an issue, we will use the opportunity to educate them
+about our process, and when necessary assist them in opening an issue. We can
+also open the issue ourselves on behalf of the user as well.
+
+### Q. Why would we re-prioritize contributions for review instead of taking them first in, first out?
+
+By default issues should be selected for review based on the stated criteria,
```

Comment:
This explanation seems a little wordy. I think we can keep it to something like:

We insertion sort new issues into the review queue to ensure that contributions that we prioritize higher (e.g. doc updates and bug fixes) can move quicker through the queue and not get blocked by potentially large, time-intensive changes.